### DISCUSSION TITLE
Setting lock strategy (and other settings) in pyptoject.toml

### DISCUSSION THREAD

**Discussion by mangunowsky** [pdm-project/pdm#2618] [Q&A] [answered]
Hello, I couldn't find anything useful on this in documentation. Here's my question:
Is it possible to offload configuration from CLI to `pyproject.toml` file? I'd like to set a lock strategy there, so developer don't have to remember about locking with parameter every time.

I'm looking for something like this at the moment, but I think having possibility to configure how PDM behaves in `pyproject.toml` would benefit coherency and limit human errors.

```
[tool.pdm.strategy]
direct_minimal_versions = true
```

Is such option live already? If not, is it possible to implement it?
Thanks!

**Comment by frostming** [comment, marked as answer]
The strategies will be encoded in the lockfile itself and be remembered, no need another place to store that right?

If your workflow creates the lockfile from scratch you can use [constant CLI options](https://pdm-project.org/latest/usage/config/#passing-constant-arguments-to-every-pdm-invocation)

**Comment by mangunowsky** [comment]
> The strategies will be encoded in the lockfile itself and be remembered, no need another place to store that right?

I remember testing it some time ago and had some issues with it, but now it is working great.
Thanks for help and for such fast response! :)
